wtxmgr/tx.go

```diff
@@ -574,6 +576,75 @@ func (s *Store) PruneUnconfirmed(height int32, stakeDiff int64) error {
 	return nil
 }
 
+// fetchAccountForPkScript fetches an account for a given pkScript given a
+// credit value, the script, and an account lookup function. It does this
+// to maintain compatibility with older versions of the database.
+func (s *Store) fetchAccountForPkScript(credVal []byte, unminedCredVal []byte,
+	pkScript []byte,
+	acctLookupFunc func(dcrutil.Address) (uint32, error)) (uint32, error) {
+	// Attempt to get the account from the mined credit. If the
+	// account was never stored, we can ignore the error and
+	// fall through to do the lookup with the acctLookupFunc.
+	if credVal != nil {
+		acct, err := fetchRawCreditAccount(credVal)
+		if err != nil {
+			storeErr, ok := err.(Error)
+			if !ok {
+				return 0, err
+			}
+			switch storeErr.Code {
+			case ErrNoExists:
+			case ErrData:
+			default:
+				return 0, err
+			}
+		}
+
+		if err == nil {
```

Comment:
For readibility, I'd move the err==nil case to the top right after the function call, and since this diverges (returns from the function), the error handling code can be unindented